be/src/storage/tablet_updates.cpp

```diff
@@ -3369,7 +3369,8 @@ Status TabletUpdates::link_from(Tablet* base_tablet, int64_t request_version, Ch
         new_rowset_info.num_segments = src_rowset.num_segments();
         // use src_rowset's meta as base, change some fields to new tablet
         auto& rowset_meta_pb = new_rowset_info.rowset_meta_pb;
-        src_rowset.rowset_meta()->get_full_meta_pb(&rowset_meta_pb);
+        // reset rowset schema to the latest one
+        src_rowset.rowset_meta()->get_full_meta_pb(&rowset_meta_pb, _tablet.tablet_schema());
```

Comment:
Why is there no tests to cover this change

Reply:
updated